build/makelib/golang.mk

```diff
@@ -162,12 +165,17 @@ go.fmt:
 
 go.validate: go.vet go.fmt
 
-.PHONY: go.vendor
-go.vendor $(GO_VENDOR_DIR)/vendor.stamp: $(GLIDE)
+$(GLIDE_LOCK): $(GLIDE) $(GLIDE_YAML)
 	@echo === updating vendor dependencies
 	@mkdir -p $(GLIDE_HOME)
+	@$(GLIDE) update --strip-vendor
+	@touch $@
```

Comment:
i understand the intent, but i don't quite understand the mechanics of how this works.  can you explain a little more how this build target will only perform the `glide update` when glide.yaml has changed?

what will be in the `$@` here when this is called?